src/components/Card.js

```diff
@@ -13,18 +26,35 @@ class Card {
 
     this._likeIcon.addEventListener("click", () => this._handleLikeIcon());
 
-    this._trashIcon.addEventListener("click", () => this._handleDeleteCard());
+    this._trashIcon.addEventListener("click", () => {
+      this._handleDeleteClick(this._id);
+      // this._handleDeleteCard();
+    });
   }
 
   _handleLikeIcon() {
     this._likeIcon.classList.toggle("card__like-button_active");
+    if (!this._likeStatus) {
+      api.likeCard(this._id);
+    }
+    api.disLikeCard(this._id);
+  }
+
+  _handleDeleteConfirm(id) {
+    deleteConfirmPopup.open(id);
   }
 
-  _handleDeleteCard() {
+  handleDeleteCard(cardId) {
     this._cardElement.remove();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.

Reply:
Corrected